src/ResourceManagement/Network/Domain/InterfaceImpl/VirtualNetworkGatewayConnectionImpl.cs

```
@@ -332,5 +332,21 @@ VirtualNetworkGatewayConnection.Definition.IWithSharedKey VirtualNetworkGatewayC
         {
             return this.WithSecondVirtualNetworkGateway(virtualNetworkGateway2) as VirtualNetworkGatewayConnection.Definition.IWithSharedKey;
         }
+
+        public IAppliableWithTags<IVirtualNetworkGatewayConnection> WithoutTag(string key)
+        {
+            return this.WithoutTag(key) as IAppliableWithTags<IVirtualNetworkGatewayConnection>;
+        }
```

remove all `as` casts. They hide real errors.